rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -148,29 +148,32 @@ class Publisher : public PublisherBase
   {
     // Topic is unused for now.
     (void)topic;
+    (void)qos;
     (void)options;
```

Comment:
Sigh.  I'm wondering whether we should deprecate this API signature, and just reduce this to:

```
virtual
void
post_init_setup(rclcpp::node_interfaces::NodeBaseInterface * node_base);
```

As far as I can tell, there are no downstream users of this API, and just continually setting this to (void) seems useless to me.  Thoughts?

Reply:
i am okay with that.

maybe we can refactor the following, but i would like to do that with another PR.

- introduce `intra_process_init` non-virtual function under `Publisher` and `Subscriber` classes, and implement whatever needs to be setup for intra-process pub/sub.
- deprecate `post_init_setup` for `Publisher` and `Subscriber` classes, `Subscriber` one is empty already. Or we can keep it for fully virtual function? so that user can override this to setup post process initialization?